flink-connectors/flink-connector-rabbitmq/src/main/java/org/apache/flink/streaming/connectors/rabbitmq/RMQSource.java

```diff
@@ -273,44 +276,56 @@ public void open(Configuration config) throws Exception {
     @Override
     public void close() throws Exception {
         super.close();
+        Exception exception = null;
 
         try {
             if (consumer != null && channel != null) {
                 channel.basicCancel(consumer.getConsumerTag());
             }
         } catch (IOException e) {
-            throw new RuntimeException(
-                    "Error while cancelling RMQ consumer on "
-                            + queueName
-                            + " at "
-                            + rmqConnectionConfig.getHost(),
-                    e);
+            exception =
+                    new RuntimeException(
+                            "Error while cancelling RMQ consumer on "
+                                    + queueName
+                                    + " at "
+                                    + rmqConnectionConfig.getHost(),
+                            e);
         }
 
         try {
             if (channel != null) {
                 channel.close();
             }
         } catch (IOException e) {
-            throw new RuntimeException(
-                    "Error while closing RMQ channel with "
-                            + queueName
-                            + " at "
-                            + rmqConnectionConfig.getHost(),
-                    e);
+            exception =
+                    ExceptionUtils.firstOrSuppressed(
+                            new RuntimeException(
+                                    "Error while closing RMQ channel with "
+                                            + queueName
+                                            + " at "
+                                            + rmqConnectionConfig.getHost(),
+                                    e),
```

Comment:
Nit: For the `channel` and `connection` you can use `IOUtils.closeAll()`

Reply:
Thanks for checking this out. Sure, I actually thought of something similar initially. I applied all suggested changes in f5564b0 - it should be cleaner now.
Should I create a back-port for 1.13 and 1.12 as well? (should be trivial)

Reply:
Yes, please also create the backports.

Reply:
PR for back-ports ready (#16377 and #16378). The only change (for both) is related to `Assert.assertThrows`. Works great here, but the previous releases depended on older junit 4.12 which did not have such a feature. I had to use a bit more cumbersome approach (using junit rules as this seems to be standard practice), but the logic remained the same.